src/tests/util.rs

```diff
@@ -511,3 +511,106 @@ pub fn generate_key_pair(
     }
     Ok((pub_key, pri_key))
 }
+
+struct CheckTemplate<'a> {
+    bytes: Vec<(CK_ATTRIBUTE_TYPE, Vec<u8>, &'a [u8])>,
+    longs: Vec<(CK_ATTRIBUTE_TYPE, CK_ULONG, CK_ULONG)>,
+    bools: Vec<(CK_ATTRIBUTE_TYPE, CK_BBOOL, CK_BBOOL)>,
+    tmpl: Vec<CK_ATTRIBUTE>,
+}
+
+pub fn check_attributes(
+    session: CK_SESSION_HANDLE,
+    handle: CK_OBJECT_HANDLE,
+    ulongs: &[(CK_ATTRIBUTE_TYPE, CK_ULONG)],
+    bools: &[(CK_ATTRIBUTE_TYPE, bool)],
+    bytes: &[(CK_ATTRIBUTE_TYPE, &[u8])],
+) -> Option<String> {
+    let capacity = ulongs.len() + bytes.len() + bools.len();
```

Comment:
nit: would be nicer to read if the parts would be sorted the same way as the arguments.
```suggestion
    let capacity = ulongs.len() + bools.len() + bytes.len();
```